### PR TITLE
Liquidation test cases

### PR DESCRIPTION
**Pull request by CalabashSquash**
### Motivation
Good idea to start writing test stubs
Also bug fixes

### Changes
- `claimReceipts` -> `claimReceipt`
- Moved some of the receipt verification from `claimReceipt` into `verifyAndSubmitLiquidation`
    - This is just because I figured all verification should go into the function with "verify" in its name
    - It might even be worth creating two functions: `verifyLiquidation` and `submitLiquidation`
- Fixed bug in `claimReceipt`, where the insurance's balance wouldn't be updated by Liquidation.sol after `drainPool` was called
- Fixed some minor bugs in `LibLiquidation.liquidationBalanceChanges`
- Added test stubs for liquidation

### CODE REVIEW

**Review by sporejack — APPROVED** [review]
LGTM!

**Review by raymogg — APPROVED** [review]

File: test/functional/Liquidation.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+const { expect } = require("chai")
+const { ethers, getNamedAccounts, deployments } = require("hardhat")
+
+describe("Liquidation functional tests", async () => {
```
Comment (sporejack) [inline test/functional/Liquidation.js:4]: It'd be nice to nest `describe` calls, like so:

```javascript
describe("Liquidation", async () => {
    describe("someLiqFunc", async () => {
        context("When called with foobar", async () => {
            it("Passes", async () => {
                /* ... */
            });
        });
    });
});
```

**Review by sporejack — CHANGES_REQUESTED** [review]

File: test/functional/Liquidation.js (hunk 1 above)
Comment (CalabashSquash) [reply test/functional/Liquidation.js:4]: I don't know if I reckon that's any better though. The output would require reading the extra "Passes" line to see what the expected result of that test case is.

File: test/functional/Liquidation.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    })
+
+    context("liquidate", async () => {
+        it("Reverts when liquidation would put liquidator below minimum margin", async () => {})
+
+        it("Reverts when agent isn't below margin", async () => {})
+
+        it("Reverts when gas price is above fast gas price", async () => {})
+
+        it("Reverts on negative liquidation amount", async () => {})
```
Comment (raymogg) [inline test/functional/Liquidation.js]: also reverts on liquidation amount = 0

File: test/functional/Liquidation.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
```
Comment (CalabashSquash) [inline test/functional/Liquidation.js]:
```suggestion

it("Reverts on liquidation amount == 0", async () => {})
```

File: test/functional/Liquidation.js (hunk 2 above)
Comment (CalabashSquash) [reply test/functional/Liquidation.js]: Added

**Pull request merged**